From 98c6c34fbbcdb556873bdcdb577fbc55a296b650 Mon Sep 17 00:00:00 2001
From: Bond_009 <bond.009@outlook.com>
Date: Tue, 20 Jun 2023 18:06:30 +0200
Subject: [PATCH] Don't ignore parentId for playlists (revert)

---
 Jellyfin.Api/Controllers/ItemsController.cs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Jellyfin.Api/Controllers/ItemsController.cs b/Jellyfin.Api/Controllers/ItemsController.cs
index e5ef6b23e98..001d8095f4f 100644
--- a/Jellyfin.Api/Controllers/ItemsController.cs
+++ b/Jellyfin.Api/Controllers/ItemsController.cs
@@ -244,7 +244,8 @@ public class ItemsController : BaseJellyfinApiController
                 .AddAdditionalDtoOptions(enableImages, enableUserData, imageTypeLimit, enableImageTypes);
 
             if (includeItemTypes.Length == 1
+                && (includeItemTypes[0] == BaseItemKind.Playlist
+                    || includeItemTypes[0] == BaseItemKind.BoxSet))
-                && includeItemTypes[0] == BaseItemKind.BoxSet)
             {
                 parentId = null;
             }
