From 99cc1ed13ac8ca3da4138500644515941f6390f6 Mon Sep 17 00:00:00 2001
From: Nyanmisaka <nst799610810@gmail.com>
Date: Mon, 25 Sep 2023 22:56:59 +0800
Subject: [PATCH] Fix A53 CC SEI breaking H26x_VAAPI hardware encode

Signed-off-by: nyanmisaka <nst799610810@gmail.com>
---
 MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs b/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs
index b6e680ab97e..c311d3b8ab9 100644
--- a/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs
+++ b/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs
@@ -44,6 +44,7 @@ public partial class EncodingHelper
         private readonly Version _minFFmpegImplictHwaccel = new Version(6, 0);
         private readonly Version _minFFmpegHwaUnsafeOutput = new Version(6, 0);
         private readonly Version _minFFmpegOclCuTonemapMode = new Version(5, 1, 3);
+        private readonly Version _minFFmpegVaapiH26xEncA53CcSei = new Version(6, 0);
 
         private static readonly string[] _videoProfilesH264 = new[]
         {
@@ -1763,6 +1764,14 @@ public string GetVideoQualityParam(EncodingJobInfo state, string videoEncoder, E
                 param += " -x265-params:0 no-info=1";
             }
 
+            /* Access unit too large: 8192 < 20880 error */
+            if ((string.Equals(videoEncoder, "h264_vaapi", StringComparison.OrdinalIgnoreCase) ||
+                 string.Equals(videoEncoder, "hevc_vaapi", StringComparison.OrdinalIgnoreCase)) &&
+                 _mediaEncoder.EncoderVersion >= _minFFmpegVaapiH26xEncA53CcSei)
+            {
+                param += " -sei -a53_cc";
+            }
+
             return param;
         }
 
